### PR TITLE
Arrow key functionality

### PR DESCRIPTION
**Pull request by brelkh**
This PR adds arrow key functionality to TAPA (ie. you can use up and down arrow keys to navigate through your previously used commands.)

This functionality was implemented as a stepping stone to the `undo` feature.

**Some limitations/notes:**
1) You can only navigate through your successfully executed commands. (If an input did not successfully execute due to typo or something, it will not appear when you use the arrow keys to look for it)
2) This extension does not "trim" your commands stored in history. If your input was `     list` (with whitespaces), while it will be trimmed before execution (ie. TAPA executes `list`), in your history, it will appear as `     list`. This shouldn't be a problem since it will still execute if it executed successfully before. 
**As an interesting followup**, in TAPA, `ListCommand` along with some others don't have a `parser` preventing you from entering other words after the `list` command. (eg. `list list list hello I am brandon` will still execute `list`) So, if that command executed successfully, using the arrow keys will make it appear the exact way the user keyed it in.
3) Once you reach the earliest command in your history, the up arrow key will move the cursor to the front of the word rather than do nothing.
4) If you close TAPA or use the `clear` command, your input history is removed and the arrow key history functionality is reset.

### CODE REVIEW

**Comment by medajet** [comment]
LGTM! Tested and working fine.

**Review by medajet — APPROVED** [review]

File: src/main/java/seedu/address/ui/CommandBox.java [hunk 1]
```diff
@@ -43,12 +51,72 @@ private void handleCommandEntered() {
...
+    private void showPreviousCommand() {
+        if (commandHistoryPointerIndex < 0) {
+            return;
+        }
+        try {
+            commandHistoryPointerIndex--;
+            String previousCommand = commandHistory.get(commandHistoryPointerIndex);
+            setCommandBoxText(previousCommand);
+        } catch (IndexOutOfBoundsException e) {
+            return; //Nothing needs to be done for requests beyond what is stored in history.
```
Comment (medajet) [inline src/main/java/seedu/address/ui/CommandBox.java:90]: Just for consistency, although it's not an important change.
```suggestion
            return; // Nothing needs to be done for requests beyond what is stored in history.
```

File: src/main/java/seedu/address/ui/CommandBox.java [hunk 2]
```diff
@@ -43,12 +51,72 @@ private void handleCommandEntered() {
...
+            setCommandBoxText("");
+            commandHistoryPointerIndex = commandHistory.size();
+            return;
+        }
+        try {
+            commandHistoryPointerIndex++;
+            String nextCommand = commandHistory.get(commandHistoryPointerIndex);
+            setCommandBoxText(nextCommand);
+        } catch (IndexOutOfBoundsException e) {
+            return; //Nothing needs to be done for requests beyond what is stored in history.
```
Comment (medajet) [inline src/main/java/seedu/address/ui/CommandBox.java:105]: Same as above.
```suggestion
            return; // Nothing needs to be done for requests beyond what is stored in history.
```

**Comment by brelkh** [comment]
> LGTM! Tested and working fine.

@medajet Thanks! I've committed your suggestions too 👍

**Pull request merged**